File: csharp/lib/Property/ScoreThresholdProperty.cs
/*
 * SPDX-FileCopyrightText: 2021-2022 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

using MessagePack;

namespace SensCord
{
    /// <summary>
    /// Structure for handling threshold of the score.
    /// </summary>
    [MessagePackObject]
    public class ScoreThresholdProperty : IBaseProperty
    {
        /// <summary>
        /// Property key.
        /// </summary>
        public static readonly string ConstKey = "score_threshold_property";

        /// <summary>
        /// Property key.
        /// </summary>
        [IgnoreMember]
        public string Key { get; } = ConstKey;

        /// <summary>
        /// Threshold of the score.
        /// </summary>
        [Key("score_threshold")]
        public float ScoreThreshold { get; set; }

        /// <summary>
        /// To string.
        /// </summary>
        /// <returns>A string that represents the current object.</returns>
        public override string ToString()
        {
            return $"{this.Key}(score_threshold={this.ScoreThreshold})";
        }
    }
}

File: java/senscord/src/senscord_properties/ScoreThresholdProperty.java
/*
 * SPDX-FileCopyrightText: 2021-2022 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package senscord_properties;

import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;

/**
 * Structure for threshold for the score to be output.
 * 
 * @see "senscord::ScoreThresholdProperty"
 */
public class ScoreThresholdProperty extends Structure {
    public float score_threshold;

    public static class ByReference extends ScoreThresholdProperty implements Structure.ByReference {
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("score_threshold");
    }

    public ScoreThresholdProperty() {
        super();
    }

    /**
     * Get score threshold value.
     * 
     * @return score threshold value.
     */
    public float GetScoreThreshold() {
        return score_threshold;
    }

    /**
     * Set score threshold value.
     * 
     * @param score_threshold score threshold value
     */
    public void SetScoreThreshold(float score_threshold) {
        this.score_threshold = score_threshold;
    }
}
